pkg/commands/server/capabilities_test.go

```diff
@@ -0,0 +1,257 @@
+	assert.Assert(t, golden.String(app.ErrBuffer.String(), "error.500.golden"))
+	test.ExpectExitCoderError(t, err)
+}
+
+// TestCapabilitiesCommandRequestSuccessYaml tests the 'capabilities' command when it gets
+// a 200 response from Synse Server, with YAML output.
+func TestCapabilitiesCommandRequestSuccessYaml(t *testing.T) {
+	test.Setup()
+
+	mux, server := test.Server()
+	defer server.Close()
+	mux.HandleFunc(
+		"/synse/2.0/capabilities",
+		func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "application/json")
+			fmt.Fprint(w, capabilitiesRespOK)
+		},
+	)
+
+	test.AddServerHost(server)
+	app := test.NewFakeApp()
+	app.Commands = append(app.Commands, ServerCommand)
+
+	err := app.Run([]string{
+		app.Name,
+		"--format", "yaml",
+		ServerCommand.Name,
+		capabilitiesCommand.Name,
+	})
+
+	t.Logf("Standard Out: \n%s", app.OutBuffer.String())
+	t.Logf("Standard Error: \n%s", app.ErrBuffer.String())
+
+	assert.Assert(t, golden.String(app.OutBuffer.String(), "capabilities.success.yaml.golden"))
+	test.ExpectNoError(t, err)
+}
+
+// TestCapabilitiesCommandRequestSuccessJson tests the 'capabilities' command when it gets
+// a 200 response from Synse Server, with JSON output.
+func TestCapabilitiesCommandRequestSuccessJson(t *testing.T) {
+	test.Setup()
+
+	mux, server := test.Server()
+	defer server.Close()
+	mux.HandleFunc(
+		"/synse/2.0/capabilities",
+		func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "application/json")
+			fmt.Fprint(w, capabilitiesRespOK)
+		},
+	)
+
+	test.AddServerHost(server)
+	app := test.NewFakeApp()
+	app.Commands = append(app.Commands, ServerCommand)
+
+	err := app.Run([]string{
+		app.Name,
+		"--format", "json",
+		ServerCommand.Name,
+		capabilitiesCommand.Name,
+	})
+
+	t.Logf("Standard Out: \n%s", app.OutBuffer.String())
+	t.Logf("Standard Error: \n%s", app.ErrBuffer.String())
+
+	assert.Assert(t, golden.String(app.OutBuffer.String(), "capabilities.success.json.golden"))
+	test.ExpectNoError(t, err)
+}
+
+// TestCapabilitiesCommandRequestSuccessPretty tests the 'capabilities' command when it gets
+// a 200 response from Synse Server, with pretty output.
+func TestCapabilitiesCommandRequestSuccessPretty(t *testing.T) {
+	test.Setup()
+
+	mux, server := test.Server()
+	defer server.Close()
+	mux.HandleFunc(
+		"/synse/2.0/capabilities",
+		func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Content-Type", "application/json")
+			fmt.Fprint(w, capabilitiesRespOK)
+		},
+	)
+
+	test.AddServerHost(server)
+	app := test.NewFakeApp()
+	app.Commands = append(app.Commands, ServerCommand)
+
+	err := app.Run([]string{
+		app.Name,
+		"--format", "pretty",
+		ServerCommand.Name,
+		capabilitiesCommand.Name,
+	})
+
+	t.Logf("Standard Out: \n%s", app.OutBuffer.String())
+	t.Logf("Standard Error: \n%s", app.ErrBuffer.String())
+
+	// FIXME: There are some strange behavior here. Tests wound't pass if there
+	// are none extra spacing in the end.
```

Comment:
interesting. this could be an issue with the formatter. you can open a separate issue to track this if you want